Gems/EMotionFX/Code/EMotionFX/Source/MotionData/MotionData.cpp

```diff
@@ -689,6 +689,12 @@ namespace EMotionFX
 
     void MotionData::ExtractRootMotion(size_t sampleJointDataIndex, size_t rootJointDataIndex, const RootMotionExtractionData& data)
     {
+        if (m_rootMotionExtracted)
+        {
+            AZ_Assert(false, "Root motion extraction already processed on this motion. Abort because run the extraction algorithm again could cause unexpected behavior.");
```

Comment:
question: Should this definitely by an `Assert` or would a `Warning` or `Error` be more appropriate here?

Reply:
https://wiki.agscollab.com/display/lmbr/Best+Practice%3A+Usage+of+AZ_Assert%2C+AZ_Error%2C+and+AZ_Warning
AZ_Assert - A programmer made a mistake and a programmer needs to fix it.
Since the current code path doesn't support applying motion extraction on motion data multiple times, this definitely worth an assert. If we do kept the original hip motion data, and we can do a revert of the data then re-do the root extraction, we could delete the assert.

Reply:
Okay makes sense thanks!